openedx/core/djangoapps/course_groups/models.py

```diff
@@ -49,3 +66,55 @@ class CourseUserGroupPartitionGroup(models.Model):
     )
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+class CourseCohortsSettings(models.Model):
+    """
+    This model represents cohort settings for courses.
+    """
+    is_cohorted = models.BooleanField(default=False)
+
+    course_id = CourseKeyField(
+        unique=True,
+        max_length=255,
+        db_index=True,
+        help_text="Which course are these settings associated with?",
+    )
+
+    cohorted_discussions = models.TextField(null=True, blank=True)  # JSON list
+
+    # pylint: disable=invalid-name
+    always_cohort_inline_discussions = models.BooleanField(default=True)
+
+
+class CourseCohort(models.Model):
+    """
+    This model represents cohort related info.
+    """
+    course_user_group = models.OneToOneField(CourseUserGroup, unique=True, related_name='cohort')
+
+    RANDOM = 'random'
+    MANUAL = 'manual'
+    ASSIGNMENT_TYPE_CHOICES = ((RANDOM, 'Random'), (MANUAL, 'Manual'),)
+    assignment_type = models.CharField(max_length=20, choices=ASSIGNMENT_TYPE_CHOICES, default=MANUAL)
+
+    @classmethod
+    def create(cls, cohort_name=None, course_id=None, course_user_group=None, assignment_type=MANUAL):
+        """
+        Create a complete(CourseUserGroup + CourseCohort) object.
+
+        Args:
+            cohort_name: Name of the cohort to be created
+            course_id: Course Id
+            course_user_group: CourseUserGroup
+            assignment_type: 'random' or 'manual'
+        """
+        if course_user_group is None:
+            course_user_group, __ = CourseUserGroup.create(cohort_name, course_id)
+
+        cls.objects.get_or_create(
+            course_user_group=course_user_group,
+            defaults={'assignment_type': assignment_type}
+        )
+
+        return course_user_group
```

Comment:
i think it's nice that the creation of a cohort is now encapsulated within one call.  I have a bad feeling, however, about the fact that `create` factory of one Model is returning an instance of a different Model.  I recognize this is because we need the cohorts API to continue to return CourseUserGroup objects, so we need to get the reference back somehow.

I can think of a few ways to deal with this
* rename the factory method (so it's less likely a future maintainer will be confused by its return value)
* move the factory method to CourseUserGroup (seems like it could work but might result in some weird dependency arrows)
* return a CouserCohort from here and adjust cohorts API to return the CUG via attribute (e.g. `return cohort.course_user_group` in add_cohort) - maybe the simplest way to go, but might have to look out for how lazily/eagerly the relationship is fetched